apps/studio/src/components/AppleLogin.tsx

```diff
@@ -0,0 +1,65 @@
+/**
+ * Logs the user into the app using the Apple Auth API.
+ */
+
+import { FunctionComponent, ReactNode } from "react";
+import {
+  AppleAuthResponse,
+  appleAuthHelpers,
+  useScript,
+} from "react-apple-signin-auth";
+import AppleIcon from "@mui/icons-material/Apple";
+import { Button } from "@newm-web/elements";
+import { VITE_APPLE_CLIENT_ID } from "@newm-web/env";
+import { setToastMessage } from "../modules/ui";
+import { useAppleLoginThunk } from "../modules/session";
+import { useAppDispatch } from "../common";
+
+interface Props {
+  readonly children?: ReactNode;
+}
+
+const AppleLogin: FunctionComponent<Props> = ({ children }) => {
+  useScript(appleAuthHelpers.APPLE_SCRIPT_SRC);
+  const dispatch = useAppDispatch();
+  const [logIn] = useAppleLoginThunk();
+
+  const redirectUri = `${window.location.origin}/apple`;
+
+  const handleLogin = () =>
+    appleAuthHelpers.signIn({
```

Comment:
This actually returns a Promise but I saw it done like this for the other logins, should I use an async/await instead? I don't think there is any issues currently as it waits for the onSuccess callback anyway.